### PR TITLE
removed glossary

### PR DESCRIPTION
**Pull request by maria-ilie**
## PR Summary 
Issue #18491 - Removed glossary file and mention of glossary in doc\contents.rst . I checked with grep and there were no other places mentioning glossary.

## PR Checklist

- [N/A ] Has pytest style unit tests (and `pytest` passes).
- [N/A ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ N/A] New features are documented, with examples if plot related.
- [N/A ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [N/A ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and `pydocstyle<4` and run `flake8 --docstring-convention=all`).
- [N/A ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [N/A ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by brunobeltran** [comment]
One of the tutorials (I think introductory/usage? don't have time to look right now) uses the glossary. If those like 2 or 3 references get changed to links this will build. 

As usual will have much more time over the weekend.

**Comment by brunobeltran** [comment]
Hi @maria-ilie, thanks for taking care of this, and sorry for the terse response earlier.

As you can see below, circleci failing to build the docs, which is because there are secretly references to it hiding in `doc/tutorials/introductory/usage.rst`, between lines 531--558-ish. As far as I can tell, this is the only file that uses these links so it would be more appropriate for the references to simply be added at the end of the file, as in the "external hyperlink targets" example here: https://docutils.sourceforge.io/docs/user/rst/quickref.html#external-hyperlink-targets

Let me know if you need any more pointers!

P.S. I recommend [building the docs](https://matplotlib.org/devel/documenting_mpl.html#getting-started) locally to make sure your changes worked correctly. If the `make html` command output ends in `make: *** [Makefile:33: html] Error 1`, then something needs to change for the docs to build correctly (`grep` for `WARNING`  in the output to find it!)

**Review by timhoffm — APPROVED** [review]

File: tutorials/introductory/usage.py [hunk 1]
```diff
@@ -454,6 +454,11 @@ def my_plotter(ax, data1, data2, param_dict):
 # .. _TkInter: https://docs.python.org/3/library/tk.html
 # .. _PyQt4: https://riverbankcomputing.com/software/pyqt/intro
 # .. _PyQt5: https://riverbankcomputing.com/software/pyqt/intro
+# .. _Qt5: https://doc.qt.io/qt-5/index.html
+# .. _GTK: https://www.gtk.org/
+# .. _Tk: http://www.tcl.tk/
+# .. _Qt4: https://doc.qt.io/archives/qt-4.8/index.html
+# .. _wxWidgets: http://www.wxwidgets.org/
```
Comment (QuLogic) [inline tutorials/introductory/usage.py:461]: These are both available securely:
```suggestion
# .. _Tk: https://www.tcl.tk/
# .. _Qt4: https://doc.qt.io/archives/qt-4.8/index.html
# .. _wxWidgets: https://www.wxwidgets.org/
```

**Comment by timhoffm** [comment]
@maria-ilie Thanks and congratulations on your first contribution to Matplotlib. We'd love to see you back!

**Comment by maria-ilie** [comment]
Thanks @timhoffm and @brunobeltran for your advice and for taking the time to review this pull request!

**Pull request merged**